### ISSUE TITLE
[FEATURE REQ] add functionality for getting help for a particular command

### ISSUE THREAD

**Issue by MadhavJivrajani** [pesos/grofer#42]
**Is your feature request related to a problem? Please describe.**
When running a particular interactive command like `grofer` or `grofer proc` it would be nice to have a way to get something like a table of available key bindings and what they do for that command. 

**Describe the solution you'd like**
Pressing a key like `h` should pause the UI rendering and display a table of the available key bindings for that command and what they do like `↑` to scroll, `k` to kill a process in the `grofer proc` command once #16 is done etc.

**Comment by anihm136** [comment]
Can I pick this up?

**Comment by MadhavJivrajani** [comment]
Absolutely!

**Comment by anihm136** [comment]
Awesome! Just to be sure, all the commands are listed in the usage section of the readme right? (i.e, no other commands/flags exist)

**Comment by MadhavJivrajani** [comment]
Yep!
All commands/flags for the latest release are available in the README.

**Comment by anihm136** [comment]
I'm still working on this :smile:

**Comment by anihm136** [comment]
I think I should wait on #54, since my changes will be in the same files

**Comment by MadhavJivrajani** [comment]
Ah okay, cool cool, I'll re-assign you

**Comment by anihm136** [comment]
So what I've been thinking for this is to create a general help widget, which can take a help string and display it as required, as well as handling the necessary keybindings. This can then be used in the 3 subcommands with the appropriate help strings. Do you think this is a good way to go? If so, what would be the best place to define such a widget?

**Comment by anihm136** [comment]
Bump @MadhavJivrajani @Gituser143 @metonymic-smokey

**Comment by MadhavJivrajani** [comment]
Hi @anihm136,
Apologies for the late response, yes that seems like a good way to go about it, just one question:
> which can take a help string and display it as required

I didn't quite get you here, could you give an example about what would count as a "help string"?

**Comment by Gituser143** [comment]
@anihm136 Sorry for the delay! This seems perfect in theory. Maybe use a textbox or table widget to make it work? You could pause the UI rendering and then render just this help widget when the help key is pressed.

If you feel that's doable, you could get started on that!

**Comment by Gituser143** [comment]
> I didn't quite get you here, could you give an example about what would count as a "help string"? 

Something like pressing `h` for help box. That's what was mentioned in the issue description...

**Comment by MadhavJivrajani** [comment]
That's what I understood from the keybindings part.

**Comment by anihm136** [comment]
@MadhavJivrajani what I mean is simply a newline-separated string, which contains some help message (need not be restricted to keybindings). The widget would handle splitting the string, formatting and displaying it appropriately. This is kind of how it's implemented in gotop, but there the help string is hardcoded into the widget, resulting in help for all the screens being displayed every time. I feel by making it modular we can be more flexible and display exactly the help that is needed for the current screen

**Comment by anihm136** [comment]
@Gituser143 Yeah a table widget is what I'm looking into. I understand the part about pausing the UI, ohpefully wont run into too many issues

**Comment by MadhavJivrajani** [comment]
Gotcha. That sounds good!

**Comment by Gituser143** [comment]
@anihm136 You might want to look into list too, that helps with scrolling!

**Comment by anihm136** [comment]
Will do

**Comment by MadhavJivrajani** [comment]
> @anihm136 You might want to look into list too, that helps with scrolling!

and also providing selection functionality based on what the user chooses while scrolling

**Comment by anihm136** [comment]
What would be the best place to define this widget, considering it's general to all the screens? I was thinking in `src/display/general`, maybe in a new file?

**Comment by MadhavJivrajani** [comment]
Maybe create a new folder `src/display/misc` or something like that, because we might need a widget/screen for killing a process, so those widgets can be grouped under that package.
